frontend/src/metabase/query_builder/components/view/PreviewQueryButton/PreviewQueryButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback } from "react";
+import { t } from "ttag";
+import Tooltip from "metabase/components/Tooltip";
+import { MODAL_TYPES } from "metabase/query_builder/constants";
+import Question from "metabase-lib/Question";
+import NativeQuery from "metabase-lib/queries/NativeQuery";
+import { PreviewButton, PreviewButtonIcon } from "./PreviewQueryButton.styled";
+
+interface PreviewQueryButtonProps {
+  onOpenModal?: (modalType: string) => void;
```

Comment:
I think it wouldn't be too hard to type `modalType` is it? I think it'd be `keyof typeof MODAL_TYPES` or something similar.

This can also be a separate discussion but I think it's better to use type in Typescript files rather than relying on calling a constant property. Ergonomics-wise, using type won't require us to import anything thing in the file, and IDE should autocomplete the parameter in `onOpenModal` automatically.